### DISCUSSION TITLE
Berry: How to Use Berry Script to Retrieve Different Endpoint Friend Names of a Zigbee Device

### DISCUSSION THREAD

**Discussion by tasict** [arendst/Tasmota#21867] [Q&A]
I understand that I can obtain the ZbName of a device through the Zigbee Class, but if different Endpoints are assigned different Names, can I use Tasmota command 'ZBStatus3' to get a list of them? Is it possible to retrieve this information using Berry Script?

**Comment by s-hadinger** [comment]
Oh you are right, retrieving endpoint details from Berry is currently not available.

It's been a long time, so I don't quite remember how to get the value with ZbInfo or ZbStatus3. It should be possible, but I'm not sure about the details. Do you have examples of ZbStatus3 with different names per endpoints? (I personally don't use this feature)

**Reply by tasict** [reply]
Currently, I am using Berry to call the Tasmota command, ZBStatus3, which returns an array of Names. This array records the Names set for all Endpoints. I am just wondering if it is possible to directly obtain the Names array using the Berry Zigbee Class alone.

**Reply by s-hadinger** [reply]
Not currently
